CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+# CMakeList.txt
+#
+# Copyright (C) 2006-2020 wolfSSL Inc.
+#
+# This file is part of wolfSSL. (formerly known as CyaSSL)
+#
+#
+
+cmake_minimum_required (VERSION 2.6)
+
+####################################################
+# Project
+####################################################
+project(wolfssl)
+find_package (Threads)
+
+####################################################
+# Compiler
+####################################################
+set(CMAKE_C_COMPILER gcc)
+set(CMAKE_CXX_COMPILER gcc)
+
+# Example for cross-compile
+#set(CMAKE_SYSTEM_NAME Linux)
+#set(CMAKE_SYSTEM_PROCESSOR arm)
+#set(CMAKE_C_COMPILER "/opt/arm-linux-musleabihf-cross/bin/arm-linux-musleabihf-gcc")
+#set(CMAKE_CXX_COMPILER "/opt/arm-linux-musleabihf-cross/bin/arm-linux-musleabihf-g++")
+#set(CMAKE_SYSROOT "/opt/arm-linux-musleabihf-cross/arm-linux-musleabihf/")
+
+####################################################
+# Build Options
+####################################################
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h")
+    # Copy generated ./options.h
+    file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h
+         DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/)
+    file(RENAME ${CMAKE_CURRENT_BINARY_DIR}/options.h
+                ${CMAKE_CURRENT_BINARY_DIR}/user_settings.h)
+else()
+   # Use template
+   file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h.in
+        DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/)
+   file(RENAME ${CMAKE_CURRENT_BINARY_DIR}/options.h.in
+               ${CMAKE_CURRENT_BINARY_DIR}/user_settings.h)
+endif()
+
+add_definitions(-DWOLFSSL_USER_SETTINGS)
+add_definitions(-DWOLFSSL_IGNORE_FILE_WARN)
+if(CMAKE_HAVE_PTHREAD_H)
+  add_definitions(-DHAVE_PTHREAD)
+endif()
+
+####################################################
+# Build flags
+####################################################
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/.)
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/.)
+
+# Example for map file and custom linker script
+set(CMAKE_C_FLAGS "-std=gnu89 ${CMAKE_C_FLAGS}")
```

Comment:
is `-std=gnu89` required? For all files? if so should check compiler support first.

Reply:
MariaDB builds without it, on at least 4 different compilers, some of which do not have this option